### PR TITLE
feat: add attach-files feature for idempotent release asset management

### PR DESCRIPTION
**Pull request by aaronsteers**
# feat: add attach-files feature for idempotent release asset management

## Summary

Implements [#20](https://github.com/aaronsteers/semantic-pr-release-drafter/issues/20): Adds a new `attach-files` input that allows users to attach build artifacts to draft releases in a single step with full idempotency.

The feature:
- Accepts file paths and glob patterns (including brace expansion like `dist/*.{whl,tar.gz}`)
- Deletes ALL existing assets before uploading new ones (ensuring idempotency even when artifact names change)
- Fails loudly if no files match the patterns or if any operation fails
- In dry-run mode, resolves and logs files that would be uploaded without making API calls

Key files:
- `lib/assets.js` - New module handling file resolution and GitHub API calls for asset management
- `index.js` - Integration point that calls asset management after release creation/update
- `action.yml` - New `attach-files` input definition
- `README.md` - Documentation for the new feature
- `test/assets.test.js` - Unit and integration tests for the new functionality

## Review & Testing Checklist for Human

- [ ] **Test in a real GitHub Actions workflow** - The GitHub API calls are mocked in tests. Create a test workflow with `attach-files` configured and verify assets are uploaded correctly to an actual draft release.
- [ ] **Test idempotency** - Run the action twice with the same files and verify the release ends up with exactly the expected assets (no duplicates).
- [ ] **Test error case** - Verify the action fails with a clear message when `attach-files` is set but no files match.
- [ ] **Consider partial failure scenario** - If deletion succeeds but upload fails partway through, the release is left in an inconsistent state. Determine if this is acceptable or needs handling.

**Recommended test plan:** Create a simple workflow that builds a test artifact, runs this action with `attach-files`, and verifies the asset appears on the draft release. Run it twice to confirm idempotency.

### Updates since last revision

- Added comprehensive test suite in `test/assets.test.js`:
  - 14 parameterized unit tests for `resolveFiles` covering glob patterns, brace expansion, deduplication, and edge cases
  - 4 integration tests for `manageReleaseAssets` covering uploads, deletions, error handling, and logging
- Added dry-run mode support: when `dry-run: true` and `attach-files` is set, the action now resolves and logs the files that would be uploaded without making API calls
- Added two integration tests to the GitHub Actions workflow matrix:
  - `attach-files-dry-run` - verifies file patterns are resolved and reported correctly
  - `attach-files-no-match-fails` - verifies the action fails with a clear error when no files match
- Fixed error handling in dry-run mode to use `core.setFailed()` instead of throwing an error, ensuring the action properly reports failure status to GitHub Actions
- Addressed Copilot feedback:
  - Changed file operations from synchronous (`fs.readFileSync`/`fs.statSync`) to async (`fs.promises.readFile`/`fs.promises.stat`) for better performance
  - Removed redundant `fs.existsSync` check (glob already filters files, and `fs.promises.readFile` will throw if file doesn't exist)
  - Added glob patterns to error messages when no files match, improving debuggability

### Notes

- Added `glob@^13.0.0` as a new dependency for pattern matching.
- File operations use async `fs.promises` API for non-blocking I/O.
- Tests use mocked GitHub API - real API behavior should still be verified manually.

**Requested by:** @aaronsteers  
**Devin session:** https://app.devin.ai/sessions/8c31f6ca523e48229d69718c48a05389

### CODE REVIEW

**Comment by aaronsteers** [comment]
Devin, If you did not create them yet, we should have a simple small parameterized set of glob unit tests.

Also, please add an integration test. It won't actually do an upload since it is in dry run mode, but in dry run mode We can at least print the list of files that we would upload.

File: lib/assets.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  await deleteAllReleaseAssets({ context, releaseId })
+
+  const uploadedAssets = []
+  for (const filePath of filesToAttach) {
+    if (!fs.existsSync(filePath)) {
+      throw new Error(`File not found: ${filePath}`)
+    }
+
+    const asset = await uploadReleaseAsset({ context, releaseId, filePath })
+    uploadedAssets.push(asset)
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply lib/assets.js]: Disagree. It is fine to leave with no assets
